src/test/java/executor/service/model/ScenarioTest.java

```diff
@@ -0,0 +1,82 @@
+package executor.service.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class ScenarioTest {
+
+    private Scenario scenario;
+
+
+    //Initialize a new script before each test
+    @Before
+    public void setUp() {
+        List<Step> steps = new ArrayList<>();
+        steps.add(new Step("Action 1", "Value 1"));
+        steps.add(new Step("Action 2", "Value 2"));
+
+        scenario = new Scenario("Test Scenario", "example.com", steps);
+    }
+
+    @Test
+    public void testGetName() {
+        assertEquals("Test Scenario", scenario.getName());
+    }
+
+    @Test
+    public void testGetSite() {
+        assertEquals("example.com", scenario.getSite());
+    }
+    @Test
+    public void testGetSteps() {
+        List<Step> expectedSteps = new ArrayList<>();
+        expectedSteps.add(new Step("Action 1", "Value 1"));
+        expectedSteps.add(new Step("Action 2", "Value 2"));
+
+        assertEquals(expectedSteps, scenario.getSteps());
+    }
+
+    @Test
+    public void testSetName() {
+        scenario.setName("New Name");
+        assertEquals("New Name", scenario.getName());
+    }
+
+    @Test
+    public void testSetSite() {
+        scenario.setSite("example1.com");
+        assertEquals("example1.com", scenario.getSite());
```

Comment:
is it good practice to test separate fields with getters when you have testEquals?